contrib/raftexample/kvstore.go

```diff
@@ -33,12 +34,15 @@ type kv struct {
 	Val string
 }
 
-func newKVStore(proposeC chan<- string, commitC <-chan *string, errorC <-chan error) *kvstore {
+func newKVStore(proposeC chan<- string, commitC <-chan *string, errorC <-chan error, ready chan struct{}) *kvstore {
```

Comment:
I think the way to initialize this after raft starts so raftNode doesn't need to know about `kv` is:
1. load snapshot (should be known where to find it independent of raft)
2. readCommits(), which replay anything that wasn't snapshotted
3. go readCommits(), to start reading in new stuff

In readCommits, it can treat `data == nil` messages as requests to save the snapshot.